src/config.rs

```diff
@@ -112,21 +109,29 @@ fn default_pdb_config_section() -> PdbConfigSection {
     }
 }
 
-#[derive(RustcDecodable,RustcEncodable,Debug)]
-pub struct CLIConfig {
-    puppetdb: Option<PdbConfigSection>,
-}
-
-impl CLIConfig {
-    fn load(path: String) -> CLIConfig {
+impl PdbConfigSection {
+    fn load(path: String) -> PdbConfigSection {
         let mut f = File::open(&path).unwrap_or_else(|e| {
             pretty_panic!("Error opening config {:?}: {}", path, e)
         });
         let mut s = String::new();
         if let Err(e) = f.read_to_string(&mut s) {
             pretty_panic!("Error reading from config {:?}: {}", path, e)
         }
-        json::decode(&s).unwrap_or_else(|e| pretty_panic!("Error parsing config {:?}: {}", path, e))
+        let json = json::Json::from_str(&s).unwrap_or_else(|e| pretty_panic!("Error parsing config {:?}: {}", path, e));
+        PdbConfigSection {
+            server_urls:
+            match json.find_path(&["puppetdb", "server_urls"])
+                .unwrap_or(&json::Json::Null).clone() {
```

Comment:
It seems like we should have a better solution here than cloning it. Any ideas here?